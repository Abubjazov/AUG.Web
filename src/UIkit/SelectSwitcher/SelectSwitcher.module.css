.root {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 210px;
  height: 40px;
  padding: 0 15px 0 10px;
  background: var(--white);
  border: none;
  border-radius: 10px;
  box-shadow: var(--box-shadow);
  animation: visible 0.5s ease-in-out;
}

.root span {
  font-size: 16px;
  font-weight: 400;
  line-height: 16px;
  color: var(--logotext);
}

@keyframes visible {
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
}
