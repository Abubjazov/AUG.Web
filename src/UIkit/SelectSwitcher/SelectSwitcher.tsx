import { FC } from 'react'

import SvgIcon from 'uikit/SvgIcon'
import { combineClasses as cc } from 'utils/combineClasses'

import styles from './SelectSwitcher.module.css'

interface SelectSwitcherProps {
  userStyles?: string
}

const SelectSwitcher: FC<SelectSwitcherProps> = ({ userStyles = '' }) => {
  return (
    <div className={cc([styles.root, userStyles])}>
      <span>Descending</span>
      <SvgIcon icon={'selectArrow'} />
    </div>
  )
}

export default SelectSwitcher
