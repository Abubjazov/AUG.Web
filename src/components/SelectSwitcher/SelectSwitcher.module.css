.root {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 270px;
  height: 40px;
  padding: 0 15px;
  background: var(--white);
  border: none;
  border-radius: 10px;
  box-shadow: var(--box-shadow);
  animation: var(--animation-visible);
}

.span-wrapper {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  cursor: pointer;
}

.root span {
  display: flex;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 10px 15px;
  font-size: 16px;
  font-weight: 400;
  line-height: 16px;
  color: var(--logotext);
  cursor: pointer;
  background: none;
  background-image: none;
}

.option:hover {
  background: var(--gray);
}

.drop-down {
  position: absolute;
  top: 43px;
  right: 0;
  z-index: 100;
  display: flex;
  flex-direction: column;
  gap: 10px;
  width: 100%;
  background: var(--white);
  border: none;
  border-radius: 10px;
  box-shadow: var(--box-shadow);
  animation: var(--animation-visible);
}

.drop-down span:hover {
  color: var(--white);
}
