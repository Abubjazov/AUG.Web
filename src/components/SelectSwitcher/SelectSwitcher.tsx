import { FC, useEffect, useState, MouseEvent } from 'react'

import { useAppDispatch, useAppSelector } from 'store/hooks'
import { EOrderBy, orderDapplets, setOrderBy } from 'store/slices/dappletsSlice'
import SvgIcon from 'uikit/SvgIcon'
import { combineClasses as cc } from 'utils/combineClasses/combineClasses'

import styles from './SelectSwitcher.module.css'

export interface SelectSwitcherProps {
  userStyles?: string
}

const SelectSwitcher: FC<SelectSwitcherProps> = ({ userStyles = '' }) => {
  const { orderBy } = useAppSelector(state => state.dapplets)

  const [isOpenedDropDown, setIsOpenedDropDown] = useState(false)

  const dispatch = useAppDispatch()

  useEffect(() => {
    dispatch(orderDapplets())
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [orderBy])

  const openCloseDropDown = () => setIsOpenedDropDown(!isOpenedDropDown)

  const changeOrderBy = (
    e: MouseEvent<HTMLSpanElement, globalThis.MouseEvent>,
  ) => {
    dispatch(setOrderBy(e.currentTarget.innerText as EOrderBy))
    openCloseDropDown()
  }

  return (
    <div className={cc([styles.root, userStyles])}>
      <div className={styles['span-wrapper']} onClick={openCloseDropDown}>
        <span
          style={orderBy ? { padding: 0 } : { padding: 0, color: '#7F7F7F' }}
        >
          {orderBy ? orderBy : 'orderBy'}
        </span>

        <SvgIcon icon={'arrowDown'} />
      </div>

      {isOpenedDropDown && (
        <div className={styles['drop-down']} onMouseLeave={openCloseDropDown}>
          <div
            className={styles['option']}
            style={{
              borderTopRightRadius: '10px',
              borderTopLeftRadius: '10px',
            }}
          >
            <span
              style={
                orderBy === EOrderBy.ASC_BY_NAME
                  ? {
                      color: '#7F7F7F',
                    }
                  : {}
              }
              onClick={changeOrderBy}
            >
              {EOrderBy.ASC_BY_NAME}
            </span>
          </div>

          <span
            className={styles['option']}
            style={
              orderBy === EOrderBy.DESC_BY_NAME ? { color: '#7F7F7F' } : {}
            }
            onClick={changeOrderBy}
          >
            {EOrderBy.DESC_BY_NAME}
          </span>

          <span
            className={styles['option']}
            style={
              orderBy === EOrderBy.ASC_BY_MARKET_CAP ? { color: '#7F7F7F' } : {}
            }
            onClick={changeOrderBy}
          >
            {EOrderBy.ASC_BY_MARKET_CAP}
          </span>

          <div
            className={styles['option']}
            style={{
              borderBottomRightRadius: '10px',
              borderBottomLeftRadius: '10px',
            }}
            onClick={changeOrderBy}
          >
            <span
              style={
                orderBy === EOrderBy.DESC_BY_MARKET_CAP
                  ? { color: '#7F7F7F' }
                  : {}
              }
              onClick={changeOrderBy}
            >
              {EOrderBy.DESC_BY_MARKET_CAP}
            </span>
          </div>
        </div>
      )}
    </div>
  )
}

export default SelectSwitcher
