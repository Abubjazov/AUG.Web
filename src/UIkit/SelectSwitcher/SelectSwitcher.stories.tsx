import type { Meta, StoryObj } from '@storybook/react'

import SelectSwitcher, { SelectSwitcherProps } from './SelectSwitcher'

import '/src/index.css'

const meta: Meta<SelectSwitcherProps> = {
  component: SelectSwitcher,
  title: 'UIkit/Select Switcher',
  parameters: {
    docs: {
      description: {
        component: 'Stub for the SelectSwitcher component',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    userStyles: {
      description: 'Passing additional custom styles for root element',
    },
  },
}

export default meta

type Story = StoryObj<SelectSwitcherProps>

export const Default: Story = {}
