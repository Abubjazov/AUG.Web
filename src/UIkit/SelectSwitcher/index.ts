import SelectSwitcher from './SelectSwitcher'

export default SelectSwitcher
