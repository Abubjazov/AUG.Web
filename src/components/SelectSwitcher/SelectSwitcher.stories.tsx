import type { Meta, StoryObj } from '@storybook/react'
import MockedProvider from 'mockData/mockedReduxProvider'

import SelectSwitcher, { SelectSwitcherProps } from './SelectSwitcher'

import '/src/index.css'

const meta: Meta<SelectSwitcherProps> = {
  component: SelectSwitcher,
  title: 'COMPONENTS/SelectSwitcher',
  parameters: {
    docs: {
      description: {
        component:
          'Interface component - SelectSwitcher. Used to configure content sorting',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    userStyles: {
      description: 'Passing additional custom styles for root element',
    },
  },
}

export default meta

type Story = StoryObj<SelectSwitcherProps>

export const Default: Story = {
  args: {},
  decorators: [
    Story => (
      <MockedProvider>
        <div style={{ height: '230px' }}>
          <Story />
        </div>
      </MockedProvider>
    ),
  ],
}
