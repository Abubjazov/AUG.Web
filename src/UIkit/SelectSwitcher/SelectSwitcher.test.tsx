import { render } from '@testing-library/react'

import SelectSwitcher from './SelectSwitcher'

describe('SelectSwitcher', () => {
  test('should render SelectSwitcher default', () => {
    const { asFragment } = render(<SelectSwitcher />)

    expect(asFragment()).toMatchSnapshot()
  })
})
