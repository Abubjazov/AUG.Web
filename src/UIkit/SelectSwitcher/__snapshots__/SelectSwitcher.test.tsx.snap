// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectSwitcher > should render SelectSwitcher default 1`] = `
<DocumentFragment>
  <div
    class="_root_50f4b1 "
  >
    <span>
      Descending
    </span>
    <div>
      <svg
        fill="none"
        height="8"
        viewBox="0 0 14 8"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 8L13.0622 0.5H0.937822L7 8Z"
          fill="#BBBBBB"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
